png.go

```diff
@@ -43,17 +43,25 @@ func (pr *pngSegmentReader) nextSegment() (r io.Reader, isMetadata bool, err err
 		return
 	}
 
-	chunkType := make([]byte, 4)
-	if _, err = io.ReadFull(teedReader, chunkType); err != nil {
-		err = &MalformedDataError{"can't parse png chunk type", err}
-		return
-	}
+	var chunkLength int64
+	if chunkDataLength == 0 {
+		// apparently some chunks have a 0 here and then an EOF after
+		chunkLength = 4
```

Comment:
Can these chunks not be metadata?

Reply:
nope - they're literally 4 bytes long, not enough to store anything

Reply:
Okay cool, thanks